blockchain/state_transition.go

```diff
@@ -376,13 +377,14 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 
 	// Defer transferring Tx fee when DeferredTxFee is true
 	if st.evm.ChainConfig().Governance == nil || !st.evm.ChainConfig().Governance.DeferredTxFee() {
-		if rules.IsMagma {
-			effectiveGasPrice := st.gasPrice
-			txFee := getBurnAmountMagma(new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), effectiveGasPrice))
-			st.state.AddBalance(st.evm.Context.Rewardbase, txFee)
+		if rules.IsKaia && msg.Type() == types.TxTypeEthereumDynamicFee {
+			st.state.AddBalance(st.evm.Context.Rewardbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), msg.EffectiveGasTip(st.evm.Context.BaseFee)))
+		} else if rules.IsMagma {
+			st.state.AddBalance(st.evm.Context.Rewardbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.gasPrice))
 		} else {
-			effectiveGasPrice := msg.EffectiveGasPrice(nil)
-			st.state.AddBalance(st.evm.Context.Coinbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), effectiveGasPrice))
+			// if EthereumDynamicFee TxType: msg.GasPrice returns msg.GasFeeCap (MaxFeePerGas)
+			// if other TxType: msg.GasPrice returns msg.Price (unitPrice)
+			st.state.AddBalance(st.evm.Context.Coinbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), msg.GasPrice()))
```

Comment:
Isn't `st.gasPrice` already EffectiveGasPrice? I didn't understand this change in non-DeferredTxFee case.

Reply:
Will be handled in Follow-up PR.